BugOsPos.Infrastructure/Persistence/DiscountRepository.cs

```diff
@@ -20,11 +20,11 @@ public Task Add(Discount discount)
         _nextId++;
         return Task.CompletedTask;
     }
-
-    public Task<Discount?> GetDiscountById(DiscountId id)
+    
+    public Task<Discount?> GetDiscountById(int id)
     {
```

Comment:
This can bring some regressive changes with other mappings that rely on `DiscountId` and not `int`.